src/main/java/com/hotels/bdp/circustrain/bigquery/context/CircusTrainBigQueryConfiguration.java

```diff
@@ -103,6 +111,20 @@ private GoogleCredentials getCredential(GCPSecurity gcpSecurity) throws IOExcept
     }
   }
 
+  @Bean
+  @Order(0)
+  CopierFactory supportedSchemeCompositeCopierFactory(
+      List<CopierFactory> copierFactories,
+      BigQueryDataExtractionManager bigQueryDataExtractionManager) {
+    BigQueryCopierFactory bigQueryCopierFactory = new BigQueryCopierFactory(bigQueryDataExtractionManager);
+    List<CopierFactory> delegates = new ArrayList<>();
+    delegates.add(bigQueryCopierFactory);
+    delegates.addAll(copierFactories);
+    //Problematic, some of these CopierFactories return true by default. Could filter here programatically but ugly + not future proof
```

Comment:
Would it be feasible to not have them return true by default but rather look at whatever params they are given and only return true if they actually match something expected? I'm not sure why the distcp copier always returns true, perhaps its intended as a fallback for any schemes that are supported by Hadoop's FileSystem API?

Reply:
I guess you could do that but its still a bit tricky and what about user defined copier plugins, unlikely this will be an issue for us personally but not good for configurability going forward

Yes, DistCpCopier returns true as it is seen as a fallback for any scheme.